chronos/transports/stream.nim

```diff
@@ -1734,19 +1753,22 @@ else:
                 trackStream(ntransp)
                 retFuture.complete(ntransp)
               else:
-                retFuture.fail(getTransportOsError(osLastError()))
+                let errorMsg = osErrorMsg(osLastError())
+                retFuture.fail(getConnectionAbortedError(errorMsg))
             else:
-              let err = osLastError()
-              if int(err) == EINTR:
+              let err = int(osLastError())
+              if err == EINTR:
                 continue
-              elif int(err) == EAGAIN:
+              elif err == EAGAIN:
                 # This error appears only when server get closed, while accept()
                 # continuation is already scheduled.
                 retFuture.fail(getServerUseClosedError())
-              elif int(err) == EMFILE:
-                retFuture.fail(getTransportTooManyError())
+              elif err in {EMFILE, ENFILE, ENOBUFS, ENOMEM}:
+                retFuture.fail(getTransportTooManyError(err))
+              elif err in {ECONNABORTED, EPERM, ETIMEDOUT}:
+                retFuture.fail(getConnectionAbortedError(err))
```

Comment:
LGTM. (have not personally seen `ETIMEDOUT` yet, but at least from the error code definition it seems to also be a case of "early close").